src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -1177,6 +1181,28 @@ private void SetRequestHandleBufferingOptions(SafeWinHttpHandle requestHandle)
             SetWinHttpOption(requestHandle, Interop.WinHttp.WINHTTP_OPTION_MAX_RESPONSE_DRAIN_SIZE, ref optionData);
         }
 
+        private void SetRequestHandleHttp2Options(SafeWinHttpHandle requestHandle, Version requestVersion)
+        {
+            Debug.Assert(requestHandle != null);
+#if NET46
+            if (requestVersion == HttpVersion20)
+#else
+            if (requestVersion == HttpVersion.Version20)
+#endif   
```

Comment:
Good idea.  I removed the ifdef's at the call sites and consolidated it to a public property.